### DISCUSSION TITLE
Is the coral worth it for me?

### DISCUSSION THREAD

**Discussion by badnbusy** [blakeblackshear/frigate#5749] [General]
I have an 8 camera 4K setup, but with detect mode at a lower res of 640x480 at 8 fps.

I am using the latest beta with openvino support on an i7 8th gen box with 16GB RAM on Debian.

Here's the system page.

https://drive.google.com/file/d/1W3YxOfOhVB81bXtrTdqz1MsP4y5nnTI2/view?usp=share_link

```
top - 17:29:40 up 2 days, 56 min,  1 user,  load average: 1.65, 1.66, 1.89
Tasks: 211 total,   2 running, 209 sleeping,   0 stopped,   0 zombie
%Cpu(s): 11.5 us,  4.1 sy,  0.0 ni, 81.8 id,  0.0 wa,  0.0 hi,  2.6 si,  0.0 st
MiB Mem :  15935.2 total,    173.1 free,   1816.4 used,  13945.7 buff/cache
MiB Swap:    977.0 total,    955.2 free,     21.8 used.  13722.8 avail Mem
```

Is the coral worth it or not? I'd ordered it just before I got openvino working on my GPU so it arrived today. Should I bother, or send it back?

Thanks

**Comment by blakeblackshear** [comment]
I don't think we have enough information to really say one way or another yet. The Coral may use less power than OpenVino on your CPU. It would also leave more room for other processing tasks. Not sure how significant that is.

**Reply by badnbusy** [reply]
Thanks. What else can I provide that would yield that information?

It may use less power or less CPU but if the current setup isn't struggling, then apart from a slightly lower Inference speed, what are the benefits?

The coral USB cost £140 so would have to cost quite a lot in extra CPU power to claim that back.

I guess the only way is to try and evaluate the difference...

**Reply by blakeblackshear** [reply]
It probably isn't worth it for you. It may be worth it if you were using a $25 Coral. We don't have any real evidence about the model accuracy yet.

**Reply by badnbusy** [reply]
Plugged in, configured, but actually seems worse at present, 30ms whereas openvino giving me 11

https://drive.google.com/file/d/1MNsjEkmm9S09ck09KhRcvZpgAXEcBl7g/view?usp=share_link

Config:

```
detectors:
  coral:
    type: edgetpu
    device: usb
```

not setting any model overrides. What gives?

**Reply by NickM-27** [reply]
How are you running frigate? USB Coral with the default model is typically 10 ms

**Reply by NickM-27** [reply]
How are you running frigate? USB Coral with the default model is typically 10 ms

**Reply by blakeblackshear** [reply]
Are you sure it's in a USB 3.0 port?

**Reply by badnbusy** [reply]
Just switched to a different USB port, now seeing:

```
Inference Speed | CPU % | Memory %
9.26 ms | 3.0% | 0.7%
```

Inference speed now 2ms lower than openvino was at 11ms. I can't see that really making any noticeable difference to detection? It is purely processing speed, or does the coral have inherent better model/detection qualities?

As I can't really tell if its any "better", not sure how I can make the decision over whether to keep it or not. Is there anything upcoming that would mean keeping it would be worth it long term?

**Reply by blakeblackshear** [reply]
It's processing speed only if the models are the same. However, the OpenVino models are different right now. I don't know if one is better than the other. Over time, I expect the same models to be available on all detector types, so the only difference would be processing speed.

**Reply by badnbusy** [reply]
Is the "CPU %" and "Memory %" shown next to the Inference speed the actual CPU in the machine or the coral/openvino detector?

**Reply by NickM-27** [reply]
> Is the "CPU %" and "Memory %" shown next to the Inference speed the actual CPU in the machine or the coral/openvino detector?

It is the CPU usage of the process running the coral, doing things like converting the image pixel format to match the model, cropping, and running the detection itself.
